### PR TITLE
translate: downgrade translation from v35 to v34

### PR DESCRIPTION
**Pull request by yasminvalim**
In this task, we need to vendor the updates from Ignition's new spec and then implement downgrade logic to transition from version 35 to version 34 and add unit tests for it. For reference, you can check the example provided https://github.com/coreos/ign-converter/pull/41.

### CODE REVIEW

File: translate/v35tov34/v35tov34.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"reflect"
+
+	"github.com/coreos/ignition/v2/config/translate"
+	"github.com/coreos/ignition/v2/config/v3_4/types"
+
+	old_types "github.com/coreos/ignition/v2/config/v3_5/types"
+	"github.com/coreos/ignition/v2/config/validate"
+)
+
+// Copy of github.com/coreos/ignition/v2/config/v3_4/translate/translate.go
```
Comment (prestist) [inline translate/v35tov34/v35tov34.go:27]: Comments are outdated need to be updated to reflect this new version

Comment (prestist) [reply translate/v35tov34/v35tov34.go:27]: to that fact, did you copy the ignition translate block for 3_5? or is this a copy from the old convert file 3_4>3_3 ?

**Review by prestist — COMMENTED** [review]
From first glance everything makes sense, I have a question, it might just be a copy pasta error, but want to verify.

File: translate/v35tov34/v35tov34.go (hunk 1 above)
Comment (yasminvalim) [reply translate/v35tov34/v35tov34.go]: I copy the transalte block for 3_5, but I copied the comments for the other file and forgot to change it. Also, I had to change the name of the last copied method. Is that a problem? 

Here we have the original: https://github.com/coreos/ignition/blob/main/config/v3_5/translate/translate.go

File: translate/v35tov34/v35tov34.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	"github.com/coreos/ignition/v2/config/v3_4/types"
+
+	old_types "github.com/coreos/ignition/v2/config/v3_5/types"
+	"github.com/coreos/ignition/v2/config/validate"
+)
+
+// Copy of github.com/coreos/ignition/v2/config/v3_5/translate/translate.go
+// with the types & old_types imports reversed (the referenced file translates
+// from 3.5 -> 3.4 but as a result only touches fields that are understood by
+// the 3.5 spec).
```
Comment (HuijingHei) [inline translate/v35tov34/v35tov34.go:30]: Maybe this should be `3.4` ?

**Review by prestist — APPROVED** [review]
Thank you for adding this, it lgtm.

**Pull request merged**